libs/test/SalesforceSDKTest/src/com/salesforce/androidsdk/rest/RestClientTest.java

```diff
@@ -633,6 +639,184 @@ public void testAccessorMethodsPreventResponseStreaming() throws Exception {
+
+        // Queries should have returned ids of newly created account and contact
+        String accountId =  jsonResults.getJSONObject(0).getJSONObject("result").getString("id");
+        String contactId =  jsonResults.getJSONObject(1).getJSONObject("result").getString("id");
+        String idFromFirstQuery = jsonResults.getJSONObject(2).getJSONObject("result").getJSONArray("records").getJSONObject(0).getString("Id");
+        String idFromSecondQuery = jsonResults.getJSONObject(3).getJSONObject("result").getJSONArray("records").getJSONObject(0).getString("Id");
+        assertEquals("Account id not returned by query", accountId, idFromFirstQuery);
+        assertEquals("Contact id not returned by query", contactId, idFromSecondQuery);
+    }
+
+    /**
+     * Test for composite request
+     *
+     * Run a composite request that:
+     * - creates an account,
+     * - creates a contact (with newly created account as parent),
+     * - run a query that should return newly created account and contact
+     *
+     * @throws IOException
+     * @throws JSONException
+     */
+    public void testCompositeRequest() throws IOException, JSONException {
+        Map<String, Object> accountFields = new HashMap<String, Object>();
+        String accountName = ENTITY_NAME_PREFIX + System.nanoTime();
+        accountFields.put("Name", accountName);
+        RestRequest firstRequest = RestRequest.getRequestForCreate(TestCredentials.API_VERSION, "account", accountFields);
+
+        Map<String, Object> contactFields = new HashMap<String, Object>();
+        String contactName = ENTITY_NAME_PREFIX + System.nanoTime();
+        contactFields.put("LastName", contactName);
+        contactFields.put("AccountId", "@{refAccount.id}");
+        RestRequest secondRequest = RestRequest.getRequestForCreate(TestCredentials.API_VERSION, "contact", contactFields);
+
+        RestRequest thirdRequest = RestRequest.getRequestForQuery(TestCredentials.API_VERSION, "select Id, AccountId from Contact where LastName = '" + contactName + "'");
+
+        SortedMap<String, RestRequest> refIdToRequests = new TreeMap<>();
+        refIdToRequests.put("refAccount", firstRequest);
+        refIdToRequests.put("refContact", secondRequest);
+        refIdToRequests.put("refSearch", thirdRequest);
+
+        // Build composite request
+        RestRequest compositeRequest = RestRequest.getCompositeRequest(TestCredentials.API_VERSION, false, refIdToRequests);
+
+        // Send composite request
+        RestResponse response = restClient.sendSync(compositeRequest);
+
+        // Checking response
+        JSONObject jsonResponse = response.asJSONObject();
+
+        JSONArray jsonResults = jsonResponse.getJSONArray("compositeResponse");
+        assertEquals("Wrong number of results", 3, jsonResults.length());
+        assertEquals("Wrong status for first request", 201, jsonResults.getJSONObject(0).getInt("httpStatusCode"));
```

Comment:
Not a big deal, but maybe use `HttpURLConnection.HTTP_CREATED` instead of `201` and `HttpURLConnection.HTTP_OK` instead of `200`, so it's easier to read?